processor/src/trace/range/mod.rs

```diff
@@ -0,0 +1,55 @@
+use super::{Felt, FieldElement};
+
+use crate::range::AuxColumnHint;
+
+pub use vm_core::range::{P0_COL_IDX, P1_COL_IDX, T_COL_IDX, V_COL_IDX};
+
+#[cfg(test)]
+mod tests;
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Builds the execution trace of the range checker's `p0` auxiliary column used for multiset
+/// checks. The running product is built up in the 8-bit section of the table and reduced in the
+/// 16-bit section of the table so that the starting and ending value are both one.
+pub fn build_aux_col_p0<E: FieldElement<BaseField = Felt>>(
+    aux_column: &mut [E],
+    aux_column_hints: &[AuxColumnHint],
```

Comment:
Maybe rather than using a slice of `AuxColumnHint` enums, we could use something like a `RangeCheckerAuxTraceHints` struct (a different name is fine too). This struct would contain a vector of `AuxColumnHint`'s but would also provide the length of the 8-bit section.

This way, we could break the single loop in the body into two separate loops - one covering the 8-bit portion and the other one covering the 16-bit portion. This would eliminate the need for `t == E::ZERO` comparisons, and also would make it easier to handle batch inversion (see my other comment).

Reply:
Sure, this makes sense, thanks